datafusion/core/src/optimizer/filter_push_down.rs

```diff
@@ -1334,6 +1337,188 @@ mod tests {
         Ok(())
     }
 
+    /// single table predicate parts of ON condition should be pushed to both inputs
+    #[ignore]
```

Comment:
This test seems to be marked `ignore`

Reply:
It is ignored, due to pushdown of ON conditions is not supported in this PR, I've created #2619 to move this logic from planner to optimizer